### ISSUE TITLE
Yojson 2.1.1 errors can have have incorrect locations

### ISSUE THREAD

**Issue by johnridesabike** [ocaml-community/yojson#171]
Example:
```ocaml
let () =
  try ignore @@ Yojson.Basic.from_string {|{"badjson": error}|}
  with Yojson.Json_error s -> print_endline s
```
Output with 2.1.0:
```
Line 1, bytes 12-18:
Invalid token 'error}'
```
Output with 2.1.1:
```
Line 1, bytes 17-18:
Invalid token 'error}'
```

**Comment by ElectreAAS** [comment]
Same problem here (example in utop)
With 2.1.0:
```ocaml
# Yojson.Safe.from_string "hello";;
Exception: Yojson.Json_error "Line 1, bytes 0-5:\nInvalid token 'hello'".
```
With 2.1.1:
```ocaml
# Yojson.Safe.from_string "hello";;
Exception:
Yojson__Common.Json_error "Line 1, bytes 4-5:\nInvalid token 'hello'".
```

**Comment by Leonidas-from-XIV** [comment]
Thanks, that's indeed a regression. Would one of you want to submit a test for it?

The only relevant change I can think of is #168, will need to be looked into.

**Comment by ElectreAAS** [comment]
> Would one of you want to submit a test for it?

On it

**Comment by Leonidas-from-XIV** [comment]
Fixed in #172.
